proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java

```diff
@@ -596,6 +601,28 @@ public void handle(Commands commands) throws Exception
         }
     }
 
+    @Override
+    public void handle(final Chat chat) throws Exception
+    {
+        ChatMessageType position = ChatMessageType.values()[chat.getPosition()];
+        Supplier<BaseComponent[]> msgSupplier = new Supplier<BaseComponent[]>() {
+            @Override
+            public BaseComponent[] get() {
+                return ComponentSerializer.parse( chat.getMessage() );
+            }
+        };
+        ServerChatEvent chatEvent = new ServerChatEvent( con.getServer(), con, msgSupplier, position );
+
+        if ( !bungee.getPluginManager().callEvent( chatEvent ).isCancelled() )
+        {
+            // Be aware that because of this, all server-sent chat messages are parsed and re-serialised
+            chat.setMessage( ComponentSerializer.toString( chatEvent.getMessage() ) );
```

Comment:
This is a problem, if you call getMessage the supply does not make any sense. This will get called everytime. Please see my suggestion above for a better idea. 

Modifying the paket should only be necessary in the cases described above.